src/main/java/org/gridsuite/study/server/service/NetworkModificationTreeService.java

```diff
@@ -776,6 +793,19 @@ public List<UUID> getStudySensitivityAnalysisResultUuids(UUID studyUuid) {
         return uuids;
     }
 
+    @Transactional(readOnly = true)
+    public List<UUID> getStudyNonEvacuatedEnergyResultUuids(UUID studyUuid) {
+        List<UUID> uuids = new ArrayList<>();
+        List<NodeEntity> nodes = nodesRepository.findAllByStudyId(studyUuid);
+        nodes.forEach(n -> {
+            UUID uuid = repositories.get(n.getType()).getNonEvacuatedEnergyResultUuid(n.getIdNode());
+            if (uuid != null) {
+                uuids.add(uuid);
+            }
+        });
+        return uuids;
+    }
```

Comment:
N+1 request

Reply:
Same thing is done for all computations ...

Reply:
Yes but we could do that easily in one custom request. I note it for another TS.